### DISCUSSION TITLE
--get-id after deployment does not give any id ?

### DISCUSSION THREAD

**Discussion by thalion0** [rustdesk/rustdesk#5038] [Q&A]
Hi Community,

while checking a possibility to silent installation with tactical rmm I mention RustDesk needs to be started manually before the CLI parameter --get-id is being able to output the ID. Steps for reproduction:

- Take a clean Win-System
- Install RustDesk silently with Installation Script: https://docs.tacticalrmm.com/3rdparty_rustdesk/

!! DO NOT START Rustdesk-GUI !!

- Open up a powershell console
-- c:\program files\rustdesk\rustdesk.exe --get-id
-- Get-Clipboard

No ID will be shown.

If you now start Rustdesk and close it - you can reproduce steps above and the ID will be shown.

Is it a bug or a planned behavior?

**Comment by dinger1986** [comment]
https://docs.tacticalrmm.com/3rdparty_rustdesk/#rustdesk-get-id-batch-collector-script-needs-custom-agent-field

Clearly says here it needs to be batch and not powershell, this should also be a script and ran as SYSTEM

Just tested myself and it isnt working now with 1.2.1 but working fine with 1.2.0

**Reply by dinger1986** [reply]
link to https://github.com/rustdesk/rustdesk/releases/tag/1.2.0

**Reply by thalion0** [reply]
Its clearly stated but it makes no difference.

**Reply by dinger1986** [reply]
what version are you installing?

**Reply by thalion0** [reply]
1.2.1

**Reply by dinger1986** [reply]
Use 1.2.0

**Reply by thalion0** [reply]
Ok, short answer: its a bug

**Reply by dinger1986** [reply]
well it isnt....cause 1.2.1 hasnt been tested yet with Tactical RMM, 1.2.0  has and works fine using the steps outlined in those docs.

However I should have said earlier that the --get-id is being reworked by the devs so presumably been taken out for now as only works with SYSTEM.

**Reply by thalion0** [reply]
I mention it has to do with the implemented encryption of id and password. Despite of using tactical as remote control or not it seems to be working only after first GUI-start of Rustdesk.

**Reply by thalion0** [reply]
But anyway. Its working and we recognized and documented the behavior. I will create a detailed report to be give as much information as possible to find the cause.

**Reply by thalion0** [reply]
Thank you so far @dinger1986

**Reply by dinger1986** [reply]
I have updated the trmm docs now https://docs.tacticalrmm.com/3rdparty_rustdesk/ thanks!

**Reply by rustdesk** [reply]
get-clipboard is a good idea

**Comment by it-boerner** [comment]
This command does not work with version 1.2.1 because it only shows the installed version of RustDesk.

$rustdesk_id = ("'C:\Program Files\RustDesk\rustdesk.exe' --get-id" | get-clipboard)
Write output $rustdesk_id

**Reply by dinger1986** [reply]
What do you mean? I tested these commands with 1.2.1

**Reply by it-boerner** [reply]
I also tested this command with RustDesk version 1.2.1 and I don't get an ID, only the display

1.2.1

**Reply by dinger1986** [reply]
In powershell?

**Reply by it-boerner** [reply]
Yes, in Powershell

**Reply by dinger1986** [reply]
this has been resolved https://github.com/rustdesk/doc.rustdesk.com/pull/239

**Reply by biz1990** [reply]
https://github.com/rustdesk/doc.rustdesk.com/pull/239 cannot resolve this problem because when run script get return value is version. Have you new method?

**Reply by dinger1986** [reply]
it works fine for me, are you running with priveleges?

**Reply by biz1990** [reply]
Right, but i want get ID only. i using c# for get ID and cannot get it. Can you suggestion idea for me?

**Reply by dinger1986** [reply]
Get it working with batch/powershell then figure out form there how to get it working with c#. What are you integrating RustDesk in to?

**Reply by biz1990** [reply]
i want to get id Rustdesk for panel control (a company software) that worker can tell for admin and admin will control computer and fix issues. Power Shell can show ID but include version message. i want get only ID but i cannot using shell get it

**Reply by dinger1986** [reply]
would you consider getting pro? if so theres maybe ways with that to get the ID

**Reply by biz1990** [reply]
because I only use it on internal system if using pro version it will not be economical
